[teozhengyang] iP

teozhengyang opened the pull request:
# 🍾 zy

> You're on your own, kid
> Yeah, you can face this
> You're on your own, kid
> You always have been -Someone famous

# Information 

zy is a **free** and personalised (to the creator only) chatbot that helps users handle and organise day-to-day tasks. ~~All the~~ Current features include:

* Create a task (ToDo/Deadline/Event)
* Mark task
* Delete task
* Search for a task

# How to run

1. Clone my [repository](https://github.com/teozhengyang/ip) to your local machine.
2. Open the project in your preferred working environment.
3. Build and run Duke.java.
If you are not sure, Duke.java contains the following ```main``` method.

```java
public static void main(String[] args) {
    new Duke("./data/duke.txt").run();
}
```

# Development Schedule

- [x] List all tasks
- [x] Mark/unmark task
- [x] Create different types of tasks
- [x] Delete task
- [x] Search task
- [x] Add GUI
- [ ] Please recommend new features 🥹

Click [here](https://www.youtube.com/watch?v=7Gbg6Z70J7E&ab_channel=TaylorSwiftVEVO) to know who's someone famous

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

teozhengyang committed:
Level-0

teozhengyang committed:
Level-1

teozhengyang committed:
Level-2

teozhengyang committed:
Level-3

teozhengyang committed:
Level-4

teozhengyang committed:
A-TextUiTesting

teozhengyang committed:
Level-5

teozhengyang committed:
Level-6 (2nd version)

teozhengyang committed:
Level-6

teozhengyang committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Duke.java

teozhengyang committed:
Add javadocs

teozhengyang committed:
Update code structure

teozhengyang committed:
Level-7

teozhengyang committed:
Merge branch 'branch-Level-7'

teozhengyang committed:
Level-8

teozhengyang committed:
Merge branch 'branch-Level-8'

teozhengyang committed:
A-MoreOOP

teozhengyang committed:
Merge branch 'branch-A-MoreOOP'

teozhengyang committed:
A-Packages

teozhengyang committed:
Merge branch 'branch-A-Packages'

teozhengyang committed:
Merge branch 'add-gradle-support'

teozhengyang committed:
A-Gradle

teozhengyang committed:
A-JUnit

teozhengyang committed:
A-Jar

teozhengyang committed:
Add JavaDoc comments

teozhengyang committed:
Merge branch 'branch-A-JavaDoc'

teozhengyang committed:
Comply with coding standard

teozhengyang committed:
Merge branch 'branch-A-CodingStandard'

teozhengyang committed:
Give users a way to find a task by searching for a keyword

teozhengyang committed:
Give users a way to find a task by searching for a keyword

teozhengyang committed:
Make code more OOP

teozhengyang committed:
Add correct import for test file

redtailedfox commented on src/main/java/duke/Parser.java:
```diff
@@ -15,96 +15,151 @@
-     * @param tasks   The TaskList object to manage tasks.
+     * @param tasks The TaskList object containing tasks to list.
+     */
+    public static void executeList(TaskList tasks) {
```
Well Done on using the proper naming convention for this function!

teozhengyang replied:
Thank you!

redtailedfox commented on src/main/java/duke/Parser.java:
```diff
@@ -3,9 +3,11 @@
 
 import java.time.LocalDate;
 import java.time.format.DateTimeParseException;
+import java.util.List;
```
Imports should be in alphabetical order

teozhengyang replied:
Thanks for pointing that out! i will be changing that in my next commit :)

teozhengyang committed:
Amend coding style violations

teozhengyang committed:
Merge branch 'branch-A-CheckStyle'

teozhengyang committed:
Add a GUI to the chatbot using JavaFX technology to implement the GUI.

teozhengyang committed:
Merge branch 'branch-Level-10'

darrentfy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,203 @@
+        String result = "";
+        try {
+            String[] separateCommand = command.split(" ");
+            if (command.equals("list")) {
+                result += Parser.executeList(tasks);
+            } else if (command.startsWith("mark") || command.startsWith("unmark")) {
+                result += Parser.executeMarkUnmark(command, tasks, separateCommand);
+            } else if (command.startsWith("todo") || command.startsWith("deadline") || command.startsWith("event")) {
+                result += Parser.executeCreate(command, tasks);
+            } else if (command.startsWith("delete")) {
+                result += Parser.executeDelete(command, tasks, separateCommand);
+            } else if (command.startsWith(("find"))) {
+                result += Parser.executeFind(command, tasks);
+            } else {
+                result += " OOPS!!! I'm sorry, but I don't know what that means :-(";
```
I like how this method is still easy to read even though there are a lot of if else checks to be done

teozhengyang replied:
thank you!

darrentfy reviewed (commented):
Overall, your code is neat, easy to read and follows the specified naming convention. Also didn't find any major problems with coding style

teozhengyang committed:
Add assert statement for improved code reliability and debugging

In this commit, I have introduced an assert statement to enhance the reliability and debugging capabilities of our Java program. This assert statement checks a critical condition at a specific point in the code, ensuring that our assumptions about the program's state hold true during runtime.

teozhengyang committed:
Enhanced Code Quality Through Abstraction

This commit focuses on improving code quality by abstracting a specific code segment into a separate function. This refactoring enhances code modularity and readability while promoting the principle of "Don't Repeat Yourself" (DRY). By isolating this code logic, we bolster maintainability and enable future reusability, reducing redundancy throughout the application. This change aligns with our commitment to delivering high-quality software, streamlining development efforts, and minimizing the risk of introducing errors. It's a proactive step toward codebase improvement, fostering cleaner, more efficient development practices.

teozhengyang committed:
Merge pull request #2 from teozhengyang/branch-A-Assertions

Add assert statement for improved code reliability and debugging

teozhengyang committed:
Merge branch 'master' into branch-A-Assertions

teozhengyang committed:
Merge branch 'master' into branch-A-CodeQuality

teozhengyang committed:
Merge branch 'branch-A-Assertions'

teozhengyang committed:
Merge pull request #3 from teozhengyang/branch-A-CodeQuality

Enhanced Code Quality Through Abstraction

teozhengyang committed:
Merge branch 'branch-A-CodeQuality'

teozhengyang committed:
Introduce Friendlier Syntax and Shorter Aliases

This commit makes the command syntax more user-friendly by introducing shorter aliases for keywords. For instance, 't' can now be used as a shorter alias for 'todo'. These changes enhance the usability of the application and provide users with more flexibility when interacting with it.

teozhengyang committed:
Revert "Introduce Friendlier Syntax and Shorter Aliases"

This reverts commit 1f9a038b3b8deccdc9ce3058959d0de39ad7a664.

teozhengyang committed:
Introduce Friendlier Syntax and Shorter Aliases

This commit makes the command syntax more user-friendly by introducing shorter aliases for keywords. For instance, 't' can now be used as a shorter alias for 'todo'. These changes enhance the usability of the application and provide users with more flexibility when interacting with it.

teozhengyang committed:
Merge branch 'branch-BCD-Extension'

teozhengyang committed:
Updated chatbot's personality with a distinct:

 Name
 Revised GUI elements including colors, icons, and fonts

teozhengyang committed:
Merge branch 'branch-A-Personality'

teozhengyang committed:
Write more JUnit tests, to test nearly all code that can be tested automatically

teozhengyang committed:
Merge branch 'branch-A-MoreTesting'

teozhengyang committed:
Add user guide to chatbot

teozhengyang committed:
Merge branch 'branch-A-UserGuide'

teozhengyang committed:
Create JAR file for chatbot

teozhengyang committed:
Merge branch 'branch-A-Release'
